compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7031,6 +7045,27 @@ private BType checkFieldAccessExpr(BLangFieldBasedAccess fieldAccessExpr, BType
         return actualType;
     }
 
+    private boolean checkRecordFieldExistence(BRecordType recordVarRefType, Name fieldName) {
+        return recordVarRefType.fields.containsKey(fieldName.getValue());
+    }
+
+    private boolean checkFieldExistenceInUnionRecordMembers(LinkedHashSet<BType> memberTypes, Name fieldName) {
+        boolean membersHaveRecordsTypes = false;
+        for (BType memberType : memberTypes) {
+            if (memberType.tag == TypeTags.RECORD) {
```

Comment:
Given that for the record tag, isSubTypeOfBaseType specifically checks for records, does this become false for any case atm?

Reply:
Updated.